flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -761,7 +784,33 @@ public JobExecutionResult execute() throws Exception {
 	 * @return The result of the job execution, containing elapsed time and accumulators.
 	 * @throws Exception Thrown, if the program executions fails.
 	 */
-	public abstract JobExecutionResult execute(String jobName) throws Exception;
+	public JobExecutionResult execute(String jobName) throws Exception {
+		if (configuration.get(DeploymentOptions.TARGET) == null) {
+			throw new RuntimeException("No execution.target specified in your configuration file.");
+		}
+
+		consolidateParallelismDefinitionsInConfiguration();
+
+		final Plan plan = createProgramPlan(jobName);
+		final ExecutorFactory executorFactory =
+				executorServiceLoader.getExecutorFactory(configuration);
+
+		final Executor executor = executorFactory.getExecutor(configuration);
+		lastJobExecutionResult = executor.execute(plan, configuration);
+		return lastJobExecutionResult;
+	}
+
+	private void consolidateParallelismDefinitionsInConfiguration() {
+		final int execParallelism = getParallelism();
+		if (execParallelism == ExecutionConfig.PARALLELISM_DEFAULT) {
```

Comment:
Shouldn't it be the other way round? That if the parallelism is not set in the `ExecutionConfig`, but set in the `Configuration` than we should update the `ExecutionConfig` with that value?

As far as I understand the parallelism setting, it is needed in the StreamGraph and Executors should not be bothered with it.